platform/security/common/src/test/java/org/codice/ddf/security/common/SPMetadataParserTest.java

```diff
@@ -0,0 +1,160 @@
+
+  @BeforeClass
+  public static void setupClass() throws Exception {
+
+    // read SPMetadata file into spMetadata
+    spMetadata = new ArrayList<>();
+    try (Stream<String> lines = Files.lines(Paths.get(FILE_PATH_2))) {
+      spMetadata.add(lines.collect(Collectors.joining(System.lineSeparator())));
+    }
+    try (Stream<String> lines = Files.lines(Paths.get(FILE_PATH))) {
+      spMetadata.add(lines.collect(Collectors.joining(System.lineSeparator())));
+    }
+
+    // set up binding set
+    bindingSet = ImmutableSet.of(Binding.HTTP_POST, Binding.HTTP_REDIRECT, Binding.SOAP);
+  }
+
+  @Test
+  public void parseNullMetadata() throws Exception {
+    Map<String, EntityInformation> testMap = SPMetadataParser.parse(null, bindingSet);
+
+    assertThat("Returns an empty map", testMap, is(equalTo(EMPTY_MAP)));
+  }
+
+  @Test
+  public void parseFirstMetadata() throws Exception {
+
+    Map<String, EntityInformation> testMap = SPMetadataParser.parse(spMetadata, bindingSet);
+
+    // get the first metadata
+    Iterator<Map.Entry<String, EntityInformation>> it = testMap.entrySet().iterator();
+    it.next();
+    Map.Entry<String, EntityInformation> testEntry = it.next();
+
+    String entityID = testEntry.getKey();
+    EntityInformation testInfo = testEntry.getValue();
+
+    assertThat("Entity ID is set", entityID, is(equalTo(ENTITY_ID)));
+    assertThat(
+        "Signing certificate is set", testInfo.getSigningCertificate(), is(equalTo(CERTIFICATE)));
+    assertThat(
+        "Encryption certificate is set",
+        testInfo.getEncryptionCertificate(),
+        is(equalTo(CERTIFICATE)));
+    assertThat(
+        "Redirect logout url is set",
+        testInfo.getLogoutService(Binding.HTTP_REDIRECT).getUrl(),
+        is(equalTo(REDIRECT_LOGOUT_URL)));
+    assertThat(
+        "Post logout url is set",
+        testInfo.getLogoutService(Binding.HTTP_POST).getUrl(),
+        is(equalTo(POST_LOGOUT_URL)));
+    assertThat(
+        "Redirect assertion consumer service url is set",
+        testInfo.getAssertionConsumerService(null, Binding.HTTP_REDIRECT, 0).getUrl(),
+        is(equalTo(REDIRECT_ACS_URL)));
+    assertThat(
+        "Post assertion consumer service url is set",
+        testInfo.getAssertionConsumerService(null, Binding.HTTP_POST, 1).getUrl(),
+        is(equalTo(POST_ACS_URL)));
+    assertThat(
+        "SOAP assertion consumer service url is set",
+        testInfo.getAssertionConsumerService(null, Binding.SOAP, 2).getUrl(),
+        is(equalTo(SOAP_ACS_URL)));
+  }
```

Comment:
✏️  Need to revise the above failure messages for all of the `assertThat` calls to help with troubleshooting.